pkg/adapter/mtping/runner.go

```diff
@@ -107,11 +107,16 @@ func (a *cronJobsRunner) RemoveSchedule(id cron.EntryID) {
 	a.cron.Remove(id)
 }
 
-func (a *cronJobsRunner) Start(stopCh <-chan struct{}) error {
+func (a *cronJobsRunner) Start(stopCh <-chan struct{}) {
 	a.cron.Start()
-	<-stopCh
-	a.cron.Stop()
-	return nil
+	<-stopCh // main channel that gets ready once term signal is received!
+}
+
+func (a *cronJobsRunner) Stop() {
+	ctx := a.cron.Stop() // no more ticks
+	if ctx != nil {      // ctx.Done has channel that gets done when all jobs completed.
```

Comment:
I don't think we need to comment about the internals of a Go context here:
```suggestion
	if ctx != nil {      // ctx gets done when all jobs complete
```